src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -12,109 +12,119 @@ import ReportActionPropTypes from './ReportActionPropTypes';
 import Clipboard from '../../../libs/Clipboard';
 import {isReportMessageAttachment} from '../../../libs/reportUtils';
 
-/**
- * A list of all the context actions in this menu.
- */
-const CONTEXT_ACTIONS = [
-    // Copy to clipboard
-    {
-        text: 'Copy to Clipboard',
-        icon: ClipboardIcon,
-        successText: 'Copied!',
-        successIcon: Checkmark,
-        shouldShow: true,
-
-        // If return value is true, we switch the `text` and `icon` on
-        // `ReportActionContextMenuItem` with `successText` and `successIcon` which will fallback to
-        // the `text` and `icon`
-        onPress: (action) => {
-            const message = _.last(lodashGet(action, 'message', null));
-            const html = lodashGet(message, 'html', '');
-            const text = lodashGet(message, 'text', '');
-            const isAttachment = _.has(action, 'isAttachment')
-                ? action.isAttachment
-                : isReportMessageAttachment(text);
-            if (!isAttachment) {
-                Clipboard.setString(text);
-            } else {
-                Clipboard.setString(html);
-            }
-        },
-    },
-
-    // Copy chat link
-    {
-        text: 'Copy Link',
-        icon: LinkCopy,
-        shouldShow: false,
-        onPress: () => {},
-    },
-
-    // Mark as Unread
-    {
-        text: 'Mark as Unread',
-        icon: Mail,
-        shouldShow: false,
-        onPress: () => {},
-    },
-
-    // Edit Comment
-    {
-        text: 'Edit Comment',
-        icon: Pencil,
-        shouldShow: false,
-        onPress: () => {},
-    },
-
-    // Delete Comment
-    {
-        text: 'Delete Comment',
-        icon: Trashcan,
-        shouldShow: false,
-        onPress: () => {},
-    },
-];
-
 const propTypes = {
     // The ID of the report this report action is attached to.
     // eslint-disable-next-line react/no-unused-prop-types
     reportID: PropTypes.number.isRequired,
 
     // The report action this context menu is attached to.
-    reportAction: PropTypes.shape(ReportActionPropTypes),
+    reportAction: PropTypes.shape(ReportActionPropTypes).isRequired,
 
     // If true, this component will be a small, row-oriented menu that displays icons but not text.
     // If false, this component will be a larger, column-oriented menu that displays icons alongside text in each row.
     isMini: PropTypes.bool,
 
     // Controls the visibility of this component.
     isVisible: PropTypes.bool,
+
+    // Function to trigger when we try to mark a message as unread
+    onMarkAsUnread: PropTypes.func,
```

Comment:
This is still a blocker for me: this should not be a prop. There's no advantage to drilling this prop down from the `ReportActionsView`. This is what we're passing down:

```javascript
() => updateLastReadActionID(this.props.reportID, item.action.sequenceNumber - 1)
```

So it's just using an Onyx action with data that's already available in this component. So instead we should just put this in the `CONTEXT_ACTIONS`.

```javascript
{
    text: 'Mark as Unread',
    icon: Mail,
    shouldShow: true,
    onPress: () => updateLastReadActionID(this.props.reportID, this.props.reportAction.sequenceNumber -1),
}
```

That removes the need for this extra prop to be drilled down through several components, and is more similar to how the other actions are being implemented.